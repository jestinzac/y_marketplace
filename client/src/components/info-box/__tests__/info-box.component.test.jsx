import { render, screen } from "@testing-library/react";

import InfoBox from "../info-box.component";

describe("InfoBox", () => {
  it("should render of type info", () => {
    const { asFragment } = render(
      <InfoBox classStyleRule="bg-blue-100" text="Test 1" />
    );

    expect(screen.getByText("Test 1")).toBeInTheDocument();
    expect(screen.getByTestId("info-box")).toHaveAttribute("data-type", "info");
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render of type error", () => {
    const { asFragment } = render(
      <InfoBox classStyleRule="bg-red-100" text="Error Text" type="error" />
    );

    expect(screen.getByText("Error Text")).toBeInTheDocument();
    expect(screen.getByTestId("info-box")).toHaveAttribute("data-type", "error");
    expect(asFragment()).toMatchSnapshot();
  });
});
