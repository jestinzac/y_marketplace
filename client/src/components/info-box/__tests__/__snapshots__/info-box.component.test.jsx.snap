// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox should render of type error 1`] = `
<DocumentFragment>
  <div
    class="block text-sm text-left h-12 flex items-center p-4 rounded-sm border bg-red-100"
    data-testid="info-box"
    data-type="error"
    role="alert"
  >
    Error Text
  </div>
</DocumentFragment>
`;

exports[`InfoBox should render of type info 1`] = `
<DocumentFragment>
  <div
    class="block text-sm text-left h-12 flex items-center p-4 rounded-sm border bg-blue-100"
    data-testid="info-box"
    data-type="info"
    role="alert"
  >
    Test 1
  </div>
</DocumentFragment>
`;
